components/gitpod-protocol/src/protocol.ts

```diff
@@ -293,11 +293,24 @@ export namespace EnvVar {
             return res;
         }
 
+        // Returns a host value, which can include port, if token also has length.
+        const getHost = (parts: string[]): string => {
+            if (parts.length === 2 && parts[0] && parts[1]) {
+                return parts[0];
+            } else if (parts.length === 3 && parts[0] && parts[1] && parts[2]) {
```

Comment:
The invariant that no part may be empty can be stuffed into a single line at the top of the function.

Reply:
Also, not entirely sure where we guard again this parttern: `" centeredKey : centeredValue "`, so it should probably check `if (parts.some(e => e.trim() === "")) { return undefined }`

Reply:
Fixed in d444551a4b93038190b9d035a5c3eea3545bfb55

Reply:
TIL about `some` function!